### PR TITLE
fix: remove nav bar back icon, use symbol/path for non-win/non-mobile

### PR DESCRIPTION
**Pull request by kazo0**
fixes https://github.com/unoplatform/uno.toolkit.ui/issues/658

Removed the BackIcon.png and MaterialBackIcon.png assets as they are no longer needed.

Uses a SymbolIcon/PathIcon for the default back button on non-windows and non-mobile platforms.

For mobile platforms: the default native back icon is used
For Windows: we cannot reuse an IconElement within a staticresource like a style (see https://github.com/microsoft/microsoft-ui-xaml/issues/1494)

### CODE REVIEW

File: src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj [hunk 1]
```diff
@@ -48,7 +48,6 @@
 
 	<ItemGroup>
 		<UpToDateCheckInput Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
-		<Content Include="Assets\**\*.png" />
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj:51]: there is no need to remove this wildcard?
same thing for v2

File: src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBar.xaml [hunk 2]
```diff
@@ -40,10 +41,10 @@
 
 	<Style x:Key="DefaultMainCommandStyle"
 		   TargetType="AppBarButton">
-		<not_mobile:Setter Property="Icon">
-			<not_mobile:Setter.Value>
-				<BitmapIcon UriSource="{StaticResource BackButtonUriSource}" />
-			</not_mobile:Setter.Value>
+		<not_mobile:Setter Property="Icon" win:Value="{x:Null}">
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBar.xaml:44]: i would add a comment here saying "the default native back icon is used"
same for the v2 counterpart

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj (hunk 1 above)
Comment (kazo0) [reply src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj:51]: We don't need it anymore since the back icon pngs were the only assets being bundled

**Comment by kazo0** [comment]
@Mergifyio backport legacy/3x

**Pull request merged**